[AY1920S1-CS2113-T14-1] Dr Duke

aquohn opened the pull request:
@gowgos5
@JacobToresson
@JeremyKwok

sanjukta99 commented on src/main/java/duke/command/ArgSpec.java:
```diff
@@ -0,0 +1,59 @@
+import java.util.Set;
+
+public abstract class ArgSpec {
+    protected String emptyArgMsg;
```
Important that you guys write JavaDoc comments for all classes and functions. You can find the Java Coding Standards for the same on the module website

sanjukta99 commented on src/main/java/duke/command/CommandHelpers.java:
```diff
@@ -0,0 +1,101 @@
+     * @return The string that the user has selected.
+     */
+    public static String disambiguate(String word, ArrayList<String> suggestions, Set<String> valid) {
+        StringBuilder builder = new StringBuilder("I didn't understand '").append(word)
```
Perhaps this method name could be made more descriptive disambiguateSwitch for instance?

sanjukta99 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,37 @@
+    public abstract void execute(DukeCore core) throws DukeException;
+
+    /**
+     * Runs the command using the parameters loaded with Command's parse method, but without any changes to storage and
```
Good use of JavaDoc comments in this file!

sanjukta99 commented on src/main/java/duke/data/Investigation.java:
```diff
@@ -0,0 +1,64 @@
+        //String resultName = "Result of " + this.getName();
+        //String resultSummary = this.summary;
+        //int resultPriority = this.getPriority
+        Result newResult = new Result(resultName, this.getImpression(), resultPriority, resultSummary);
```
Commented out code should be deleted

sanjukta99 commented on src/main/java/duke/data/Patient.java:
```diff
@@ -0,0 +1,220 @@
+    private String bedNo;
+    private String allergies;
+    private Impression priDiagnosis;
+    private ArrayList<Impression> impressions;
```
Giving the variable `priDiagnosis` a more descriptive name like `primaryDiagnosis` might convey the use of the variable better. Let's someone who is reading your code for the first time know the purpose of that variable

sanjukta99 commented on src/main/java/duke/data/Tsk.java:
```diff
@@ -0,0 +1,95 @@
+ * files.
+ */
+public enum Tsk {
+    TODO("T") {
```
Should give this enum a more descriptive name. `Tsk` conveys nothing about the purpose of the enum

sanjukta99 commented on src/main/java/duke/command/ArgCommand.java:
```diff
@@ -0,0 +1,70 @@
+    protected abstract ArgSpec getSpec();
+
+    // I hate Java
+
```
:)

sanjukta99 reviewed (commented):
Good work so far! Take a look at my suggestions. Keep in mind the Java Coding Standards, especially for JavaDoc comments. While I can see that you guys have attempted to write comments, please try to be consistent and follow the format. You guys could also try to use more descriptive names for variables (even though they may make the variable name longer). This is so that anyone who is reading your code is immediately able to understand what is the purpose of the code.

aquohn committed:
Merge branch 'list-model'

aquohn committed:
Use open to change contexts rather than setContext

aquohn committed:
Abstract debug string comparison function and comment out failing tests

aquohn committed:
Setup loggers

aquohn committed:
Log distance array

aquohn committed:
Add new parser tests

aquohn committed:
Move autocorrect code to 'unused'

aquohn committed:
Begin

aquohn committed:
Update

aquohn committed:
Update Impression edit

gowgos5 committed:
Merge pull request #305 from aquohn/master

Use ObjCommand for search

gowgos5 committed:
Build successfully.

gowgos5 committed:
Add DukeData context windows

gowgos5 committed:
Fix codacy issues

gowgos5 committed:
Fix codacy issues

gowgos5 committed:
Patient tests are up

gowgos5 committed:
Add tests for UiContext

aquohn committed:
Merge pull request #306 from gowgos5/branch-clean

Add DukeData context windows

aquohn committed:
Create ImpressionEdit command tests

aquohn committed:
Fix codacy violations

aquohn committed:
Perform minor cleanup

gowgos5 committed:
Bug fixes

gowgos5 committed:
Merge pull request #307 from aquohn/master

Create ImpressionEdit command tests

gowgos5 committed:
Merge remote-tracking branch 'upstream/master' into branch-clean

aquohn committed:
Add indexed lists to search

aquohn committed:
Merge pull request #308 from gowgos5/branch-clean

Add JUnit tests for UIContext and Patient classes

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

aquohn committed:
Update back and up command descriptors

aquohn committed:
Fix bug in Patient findAll method

aquohn committed:
Move storage classes to storage package

aquohn committed:
Create and update contains methods in patient and impression

aquohn committed:
Move summary field down to individual DukeData classes

aquohn committed:
Update DukeData with contains method

aquohn committed:
Change SearchResult to use instanceof

aquohn committed:
Replace \n with lineSeparator before printing

aquohn committed:
Remove home history command

aquohn committed:
Fix HomeReport

aquohn committed:
Update PatientTest to deal with lineseparator

gowgos5 committed:
Updated user guide

aquohn committed:
Merge pull request #309 from gowgos5/branch-ug

Updated user guide

aquohn committed:
Add Impression Command tests

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

aquohn committed:
Fix newline bug

aquohn committed:
Update open mesage

aquohn committed:
Update Impression edit to handle unambiguous data

aquohn committed:
Create isComplete flags for ObjCommand

aquohn committed:
Add persistent storage of queued command

gowgos5 committed:
Change font size of input textfield

gowgos5 committed:
Tidy code for FXML files

aquohn committed:
Add Impression move tests and create getCmd method in argspec and objspec

aquohn committed:
Fix failing move tests

aquohn committed:
Create ObjCommand tests for open, and the objcommand system

aquohn committed:
Create ImpressionOpen command

aquohn committed:
Fix checkstyle errors

gowgos5 committed:
Merge pull request #310 from aquohn/master

Bugfixes, refactoring, tests and ImpressionOpen

gowgos5 committed:
Refactor Ui classes

gowgos5 committed:
Merge remote-tracking branch 'upstream/master' into branch-ui

aquohn committed:
Update Home context up/back messages

aquohn committed:
Create skeletons for patient and impression open tests

aquohn committed:
Remove deletePrimaryDiagnosis

aquohn committed:
Create priority constants in DukeData

aquohn committed:
Create isFollowUp method

aquohn committed:
Update add and delete methods for patient to maintain critical and follow up lists

aquohn committed:
Update search by name in patient to use lists

gowgos5 committed:
More refactoring

aquohn committed:
Create DukeObject update method

aquohn committed:
Remove PatientContextWindow dependency on indexed lists

aquohn committed:
Remove PatientUtils dependence on indexed lists

aquohn committed:
Update patient commands to refer to followups instead of investigations

gowgos5 committed:
Merge pull request #311 from aquohn/master

Remove indexed list dependency

gowgos5 committed:
Refine user guide

gowgos5 committed:
Merge remote-tracking branch 'upstream/master' into branch-ui

aquohn committed:
Merge pull request #312 from gowgos5/branch-ui

DukeData context windows up

aquohn committed:
Create DukeData edit commands

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

aquohn committed:
Call object update from Ui update

aquohn committed:
Update ResultEdit and fix PatientPrimary

aquohn committed:
Fix fatal errors in latest PR

aquohn committed:
Remove Travis badge which is negatively affecting morale

gowgos5 committed:
Merge pull request #313 from aquohn/master

Essential fix

gowgos5 committed:
Removed getIndexedList()

gowgos5 committed:
Merge remote-tracking branch 'upstream/master' into branch-ui

aquohn committed:
Factor out data editing and complete ResultEditSpec

gowgos5 committed:
hot fix

aquohn committed:
Add remaining edit commands

JeremyKwok committed:
SearchTest: PatientData & Patient

aquohn committed:
Minor cleanup

aquohn committed:
Merge pull request #314 from gowgos5/branch-ui

Removed getIndexedList()

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

aquohn committed:
Merge pull request #315 from gowgos5/branch-ui

hot fix

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

aquohn committed:
PatientObjSpec fix

JeremyKwok committed:
pull updates from master
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-1/main into SearchTest

JeremyKwok committed:
Checkstyle fixes

aquohn committed:
Remove updateContext

aquohn committed:
Updates to Ui systems

JeremyKwok committed:
NPE fix

gowgos5 committed:
Merge pull request #316 from JeremyKwok/SearchTest

Search test

aquohn committed:
Perform input validity check for data before asking for impression

gowgos5 committed:
Fix DukeData context windows

aquohn committed:
Refactor ImpressionMove to abort if parameters are invalid before searching

aquohn committed:
Bugfixes for ImpressionMove

aquohn committed:
Fix Observation context

aquohn committed:
Fix bugs found through tests

aquohn committed:
Remove Investigation toString()

aquohn committed:
Merge pull request #317 from gowgos5/branch-ui

Fix DukeData context windows

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

aquohn committed:
Complete ObjCommand test

aquohn committed:
Add corner cases to ImpressionMoveTest

aquohn committed:
Fix cCheckstyle

gowgos5 committed:
Updated UG

gowgos5 committed:
Merge pull request #318 from aquohn/master

Fixes and refactors

gowgos5 committed:
Update user guide

aquohn committed:
Merge pull request #319 from gowgos5/branch-ug-updated

Updated UG

aquohn committed:
Revert "Remove Travis badge which is negatively affecting morale"

This reverts commit 742eebbf20458c8760d8fec38ef885eaf1fe2907.

aquohn committed:
Add status, priority and delete tests for Impression context

JeremyKwok committed:
Test criticality and follow up counting

JeremyKwok committed:
test finding for impressions

aquohn committed:
Complete Impression Result command test

aquohn committed:
Merge pull request #320 from JeremyKwok/SearchTest

Search test: Impression and ImpressionTest:Counts

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

aquohn committed:
Add Find command tests

aquohn committed:
Check for duplicates when combining searchresults

aquohn committed:
Fix Codacy violations

aquohn committed:
Add observation to impression new command test

aquohn committed:
Refactor DukeData to use setters in constructor

JeremyKwok committed:
Update model diagram

JeremyKwok committed:
Merge pull request #321 from JeremyKwok/SearchTest

Update model diagram

aquohn committed:
Add patient delete command

aquohn committed:
Setup class variables for patient command tests

aquohn committed:
Allow patient name and bed number to be set

aquohn committed:
Add Patient Edit test

aquohn committed:
Add Patient primary test

aquohn committed:
Restore disabled storage tests

aquohn committed:
Fix GsonStorage test

aquohn committed:
Add required data files

gowgos5 committed:
Updated UG

aquohn committed:
Fix checkstyle violations

aquohn committed:
Merge pull request #322 from gowgos5/branch-ug-updated

Updated UG

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

gowgos5 committed:
Merge pull request #323 from aquohn/master

Add more tests

aquohn committed:
Update storage tests to handle line separators

aquohn committed:
Remove magic constants

aquohn committed:
Refactor priority string computation to be in dukedata

aquohn committed:
Remove patientCountStr function

aquohn committed:
Cleanup and handle TODOs

aquohn committed:
Fix test errors

aquohn committed:
Another lineseparator fix

aquohn committed:
Revert "Revert "Remove Travis badge which is negatively affecting morale""

This reverts commit bf17bba5595c486fd39f5e0f4ed5eff455805cf8.

aquohn committed:
Merge pull request #324 from aquohn/master

Update storage tests to handle line separators

gowgos5 committed:
Tidy up UI and write Javadocs for Junit tests

gowgos5 committed:
Merge remote-tracking branch 'upstream/master' into branch-ui

aquohn committed:
Cleanup and add @@author tags

aquohn committed:
Add @@author tags

gowgos5 committed:
Fix codacy issues

aquohn committed:
Update author tags

aquohn committed:
Test opening of primary impression

aquohn committed:
Fix checkstyle violations

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

gowgos5 committed:
Merge pull request #326 from aquohn/master

Author tags and cleanup

gowgos5 committed:
Merge remote-tracking branch 'upstream/master' into branch-ui

aquohn committed:
Update UG

aquohn committed:
Merge pull request #325 from gowgos5/branch-ui

Javadocs for tests

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

JeremyKwok committed:
Model Documentation

gowgos5 committed:
Package data into jar

gowgos5 committed:
Merge pull request #327 from aquohn/master

Update UG

gowgos5 committed:
Mr. Travis

gowgos5 committed:
Merge pull request #328 from JeremyKwok/documentation

DG: Model Documentation

gowgos5 committed:
Merge remote-tracking branch 'upstream/master' into branch-ui

aquohn committed:
Minor bugfixes

aquohn committed:
Merge pull request #329 from gowgos5/branch-ui

Misc stuff

gowgos5 committed:
Update README.md

JeremyKwok committed:
update DG feature find

gowgos5 committed:
Final UG

gowgos5 committed:
Merge pull request #330 from JeremyKwok/documentation

update DG feature find

gowgos5 committed:
Merge remote-tracking branch 'upstream/master' into branch-ui

gowgos5 committed:
Merge pull request #331 from gowgos5/branch-ui

Final UG

gowgos5 committed:
Update UserGuide.adoc

gowgos5 committed:
Removed ui

gowgos5 committed:
Add UI

gowgos5 committed:
Restore test

gowgos5 committed:
Merge remote-tracking branch 'upstream/master' into branch-ui

gowgos5 committed:
Merge pull request #332 from gowgos5/branch-ui

UG

gowgos5 committed:
Remove UI

gowgos5 committed:
Add UI

gowgos5 committed:
Merge pull request #333 from gowgos5/branch-ui

Branch ui

JeremyKwok committed:
Added photo for dg

JeremyKwok committed:
wrong case for file

JeremyKwok committed:
Author tags

aquohn committed:
Add sequence diagram

JeremyKwok committed:
Merge pull request #334 from JeremyKwok/documentation

Added photo for dg

aquohn committed:
Update DG parser logic section

aquohn committed:
Update DG parser section

aquohn committed:
Update FSM section of DG

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

gowgos5 committed:
Refine DG

gowgos5 committed:
Merge pull request #335 from gowgos5/branch-ui

Refine DG

JeremyKwok committed:
Fix language issues in DG

JeremyKwok committed:
Merge pull request #336 from JeremyKwok/documentation

Fix language issues in DG

gowgos5 committed:
Update AboutUs.adoc

gowgos5 committed:
Update AboutUs.adoc

gowgos5 committed:
Update README.md

JeremyKwok committed:
added instruction for testing search

aquohn committed:
Add diagram to Parser section

JeremyKwok committed:
Merge pull request #337 from JeremyKwok/documentation

added instruction for testing search

JeremyKwok committed:
Add uml diagram for search function

JeremyKwok committed:
Merge pull request #338 from JeremyKwok/documentation

Add uml diagram for search function

gowgos5 committed:
Update README.md

aquohn committed:
Update ObjCommand section

JeremyKwok committed:
Added additional UML diagrams

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

JeremyKwok committed:
update test cases

JeremyKwok committed:
Merge pull request #339 from JeremyKwok/documentation

Added additional UML diagrams

JeremyKwok committed:
add desgin considerations for DataModel

JeremyKwok committed:
Merge pull request #340 from JeremyKwok/documentation

add desgin considerations for DataModel

aquohn committed:
Update DG and add manual testing instructions

aquohn committed:
Complete sequence diagrams

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

aquohn committed:
Merge pull request #341 from aquohn/master

Merges and fixes

aquohn committed:
Update

aquohn committed:
Fix failing tests

gowgos5 committed:
Merge pull request #342 from aquohn/master

Update DG and add PPP

gowgos5 committed:
Update DG

gowgos5 committed:
Merge remote-tracking branch 'upstream/master' into branch-ui

aquohn committed:
Move PPP to proper folder

aquohn committed:
Update PPP

gowgos5 committed:
Finalise docs

gowgos5 committed:
Merge pull request #343 from gowgos5/branch-ui

Finalise documentation

aquohn committed:
Update PPP

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

aquohn committed:
Merge pull request #344 from aquohn/master

PPP changes

gowgos5 committed:
Update AboutUs.adoc

JeremyKwok committed:
AuthorTags for Impresion

gowgos5 committed:
Update vernon.adoc

aquohn committed:
Attempt to compress PPP

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

JeremyKwok committed:
Add PPP

JeremyKwok committed:
update ppp link

gowgos5 committed:
Final push?

gowgos5 committed:
Merge remote-tracking branch 'upstream/master' into branch-ui

gowgos5 committed:
Merge pull request #345 from gowgos5/branch-ui

Final commit?

aquohn committed:
Merge pull request #346 from aquohn/master

DG update

aquohn committed:
Fix DG bug

gowgos5 committed:
v1.4

gowgos5 committed:
Merge remote-tracking branch 'upstream/master' into branch-ui

aquohn committed:
Final update

aquohn committed:
Merge branch 'master' of github.com:AY1920S1-CS2113-T14-1/main

aquohn committed:
Merge pull request #347 from gowgos5/branch-ui

Final PR

aquohn committed:
Merge pull request #348 from aquohn/master

Final PR
